beacon-chain/powchain/rpc_connection.go

```diff
@@ -0,0 +1,166 @@
+package powchain
```

Comment:
This file consolidates connection management functions and refactors for nicer separation of concerns